vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -84,77 +84,103 @@ public ImageData getImageDimensions(IVSImage image, int width, int height) {
+    public List<IVSImage> getImages(int pageNo, String category) {
+        return getImages(pageNo, category, SortByField.CREATION_DATE.getValue(), Sort.Direction.DESC.toString());
+    }
+    
+    
     @Override
     public List<IVSImage> getImages(int pageNo) {
-        return getImages(pageNo, SortByField.CREATION_DATE.getValue(), Sort.Direction.DESC.toString());
+        return getImages(pageNo, null, SortByField.CREATION_DATE.getValue(), Sort.Direction.DESC.toString());
     }
-
+    
     /**
      * Method to return the requested images
      * 
      * @param pageNo. if pageNo<1, 1st page is returned, if pageNo>total pages,last
      *                page is returned
      * @return list of images in the requested pageNo and requested order.
      */
+
     @Override
-    public List<IVSImage> getImages(int pageNo, String sortedBy, String order) {
-        Sort sortingParameters = Sort.by(SortByField.CREATION_DATE.getValue()).descending();
-        pageNo = validatePageNumber(pageNo);
-        if(sortedBy!=null && SortByField.getAllValues().contains(sortedBy)) {
-            sortingParameters = Sort.by(sortedBy);
-        }
-        if(order!=null && order.equalsIgnoreCase(Sort.Direction.ASC.toString())) {
-            sortingParameters = sortingParameters.ascending();
-        }
-        else {
-            sortingParameters = sortingParameters.descending();
-        }
+    public List<IVSImage> getImages(int pageNo, String category, String sortedBy, String order) {
+        Sort sortingParameters = getSortingParameters(sortedBy, order);
+        pageNo = validatePageNumber(pageNo, category);
         Pageable sortByRequestedField = PageRequest.of(pageNo - 1, pageSize, sortingParameters);
-        Page<VSImage> images = imageRepo.findAll(sortByRequestedField);
+        Page<VSImage> images;
+        if(category==null || category.equals("null") || category.isEmpty()) {
+            images = imageRepo.findAll(sortByRequestedField);
+        } else {
+            images = imageRepo.findByCategories(sortByRequestedField, ImageCategory.valueOf(category));
+        }
         List<IVSImage> results = new ArrayList<>();
         if(images != null) {
             images.getContent().forEach(i -> results.add(i));
         }
         return results;
     }
-
+    
     /**
-     * Method to return the total pages sufficient to display all images
+     * Method to return the total image count
      * 
-     * @return totalPages required to display all images in DB
+     * @return total count of images in DB
      */
+
     @Override
-    public long getTotalPages() {
-        return (imageRepo.count() % pageSize==0) ? imageRepo.count() / pageSize:(imageRepo.count() / pageSize) + 1;
+    public long getTotalImageCount(String category) {
+        if(category==null || category.equals("null") || category.isEmpty()) {
+            return imageRepo.count();
+        }
+        return imageRepo.countByCategories(ImageCategory.valueOf(category));
```

Comment:
also, just noticing, I would pass in an ImageCategory and making the conversion from string to ImageCategory before calling this class.

Reply:
In that case, we would need to add the "All" constant to handle all images without any filter applied scenario.

Reply:
or just null?